### ISSUE TITLE
Not possible to change accountEnabled with application permissions on non-windows devices

### ISSUE THREAD

**Issue by rstenborg** [microsoftgraph/microsoft-graph-docs#16641]
Hi! 

When doing a PATCH against https://graph.microsoft.com/v1.0/devices/AADdeviceID with content: {accountEnabled: false} against a device that is not a Windows device (for example an iPad) you get this error: "Properties other than ExtendedAttribute1..15 can be modified only on windows devices."

The above call works as intended when you run it as a signed in user (for example through Graph Explorer), but it doesnt work when running with application permissions (as a service principal). 

This is a bummer since I need this to work with application permissions in order to automate the workflow of disabling stale AAD devices. 

Is this by design? If it is, it should be clarified in the docs.

Thanks! 

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: a2823c35-1b67-2d65-1585-0c6d6162ef8e
* Version Independent ID: c10ea5e9-24d2-910c-5c10-58cc2188fc1f
* Content: [Update device - Microsoft Graph v1.0](https://docs.microsoft.com/en-us/graph/api/device-update?view=graph-rest-1.0&tabs=http)
* Content Source: [api-reference/v1.0/api/device-update.md](https://github.com/microsoftgraph/microsoft-graph-docs/blob/main/api-reference/v1.0/api/device-update.md)
* Product: **directory-management**
* Technology: **microsoft-graph**
* GitHub Login: @SanDeo-MSFT
* Microsoft Alias: **MSGraphDocsVteam**

**Comment by bashuta** [comment]
> This is a bummer since I need this to work with application permissions in order to automate the workflow of disabling stale AAD devices.
> 
> Is this by design? If it is, it should be clarified in the docs.

The same problem and exactly the same task here.  Doesn't appear to be "by design" as [Graph API - Update device](https://docs.microsoft.com/en-us/graph/api/device-update?view=graph-rest-1.0&tabs=http#permissions) shows both the very example and the application permissions. Neither permission works for an application.

**Comment by rstenborg** [comment]
I also created a support case regarding this. I'll update this issue accordingly.

**Comment by FaithOmbongi** [comment]
Thank you @rstenborg, @bashuta for raising this issue. Please assign the app the **Cloud Device Administrator** role as per the following note in the property description.

<img width="536" alt="image" src="https://user-images.githubusercontent.com/14026935/165722857-e3ee1040-ba99-4ee1-bf1c-493b2b6b6fbf.png">

Repro:
**Permission minus the Cloud Device Administrator role**

| Token snapshot| Response|
|:--- |:--- |
| **Permissions minus role** <img width="332" alt="image" src="https://user-images.githubusercontent.com/14026935/165721014-a27ecccd-0b06-4b20-a8ce-44e3fcb58d86.png">|<img width="386" alt="image" src="https://user-images.githubusercontent.com/14026935/165722245-9733a5c8-0a89-4c81-924e-7b40f63ea42f.png"> |
| **Permissions and Cloud Device Administrator role**<img width="332" alt="image" src="https://user-images.githubusercontent.com/14026935/165721766-80b16a9d-636e-4ff6-9c3f-765802a54aab.png"> | <img width="450" alt="image" src="https://user-images.githubusercontent.com/14026935/165721422-b4f3688c-7f87-4181-8b0c-f2d19d3b220a.png"> |

Let me also raise a PR to update the Response body section of the [Update device](https://docs.microsoft.com/en-us/graph/api/device-update?view=graph-rest-1.0&tabs=http) topic so this requirement is more visible.

**Comment by FaithOmbongi** [comment]
Thank you @rstenborg, @bashuta for raising this issue. Let me investigate and get back to you.

**Comment by rstenborg** [comment]
Just got an update on my support case as below. Hopefully it will be resolved soon. 

_Just got a confirmation from the PM that this behaviour is not intended and they are going to fix this issue. However, as I was expecting, there is no ETA to share right now. Since this is affecting v1.0, hopefully they can add it as a priority, but my guess is that it will take some months.

I would like to give you a better answer but unfortunately this is what I can give you for now.

If you want a fully working solution right now, the best approach will really be to set up a ROPC flow, as I mentioned previously._

**Comment by FaithOmbongi** [comment]
Thank you everyone for keeping this conversation going. I've confirmed the message that @rstenborg has shared - the product team is working to resolve this issue but we don't have an ETA to share.

I recommend that you file a [feature request](https://feedbackportal.microsoft.com/feedback/forum/ebe2edae-97d1-ec11-a7b5-0022481f3c80) to amplify your voices on this request. One person can file the feature request and post the link to that feature request back here so others can amplify their voice by upvoting the feature request.

Closing this docs issue.
